### PR TITLE
fix: webhook process

### LINKED ISSUE #45345: Users can continue using CodeAlley while they are singed off
@Tom_M After users sign out of their account, CodeAlly related data remains in the local Storage and CodeAlly challenges could be accessed and continued. Not sure if this is the intended behavior or not.

### PR DESCRIPTION
**Pull request by moT01**
Does:
-this moves the showIframe state that was in the component to redux
-tokens now get created when clicking the start project button
-tokens get deleted when signing out, resetting account, deleting account
-can delete token from settings page
-renames webhookToken to userToken

Doesn't do:
~~-There's no longer a way for users to manually delete or renew their tokens. Do we want to have that ability?~~ added^^
-Possibly want to add a loader after users click 'start' so they don't click around while their token is being created.
~~-If we want to rename 'webhook token' to something else, now would be the time - before we create the collection on production. I wouldn't mind doing that. I was thinking 'user token'. 'auth token' could be confused with the other auth token? Pretty minor.~~ added^^

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #45345

### CODE REVIEW

**Comment by raisedadead** [comment]
> There's no longer a way for users to manually delete or renew their tokens. Do we want to have that ability?


That's a good question. I think we give them the ability in the settings page, just as a troubleshooting measure?

File: client/src/utils/tone/index.ts [hunk 1]
```diff
@@ -21,8 +21,6 @@ const toneUrls = {
   [FlashMessages.CertificateMissing]: TRY_AGAIN,
   [FlashMessages.CertsPrivate]: TRY_AGAIN,
   [FlashMessages.CompleteProjectFirst]: TRY_AGAIN,
-  [FlashMessages.CreateTokenErr]: TRY_AGAIN,
-  [FlashMessages.DeleteTokenErr]: TRY_AGAIN,
```
Comment (ojeytonwilliams) [inline client/src/utils/tone/index.ts:25]: I think we can use this, since we're allowing users to manually delete tokens.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
> -If we want to rename 'webhook token' to something else, now would be the time - before we create the collection on production. I wouldn't mind doing that. I was thinking 'user token'. 'auth token' could be confused with the other auth token? Pretty minor.

Yep, I think we should.

**Review by ojeytonwilliams — DISMISSED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Comment by nhcarrigan** [comment]
Everything on our end is looking good - just one thing @moT01:

- I might be missing something, but I can't seem to get a CodeRoad container to pick up a new token. When I `printenv` it shows the old one still.

**Comment by moT01** [comment]
You have to wait five minutes for your container to shut down. Then when it starts back up, it should have the new one. Perhaps you tried that, and there is an issue. Since you posted that an hour ago, it should work now. Care to give it another try and check again @nhcarrigan?

How to tell the difference between running/not running VM's when they load:
-A container that is shut down will go through all the loading steps [in this picture](https://forum.freecodecamp.org/t/how-to-troubleshoot-the-web-version-of-the-relational-database-curriculum/500231#you-are-stuck-on-the-last-step-of-loading-remote-environment-3)
-A running container will only go through the steps after that screen. It's just a couple steps - maybe something like 'finalizing load' or 'loading virtual machine'

**Comment by nhcarrigan** [comment]
I'll give it another shot - I did wait a while before trying, and saw the loading screen. But my container opened with the same terminal open I had before.

**Review by naomi-lgbt — APPROVED** [review]
Triaged the token regen issue with Tom, it's not an issue on our side it's related to CodeRoad. Shouldn't block this PR so I think we're good to go here.

**Pull request merged**